mne/preprocessing/annotate_nan.py

```diff
@@ -0,0 +1,41 @@
+# Author: David Julien <david.julien@ifsttar.fr>
+#
+# License: BSD (3-clause)
+
+from ..annotations import (Annotations)
+
+
+def annotate_nan(raw):
+    """Detect segments with NaN and annotate them.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to find segments with NaN values.
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with NaN segments annotated as NAN
+    """
+    annot = raw.annotations
+
+    onsets = []
+    durations = []
+    desc = []
+    sampling_duration = 1 / float(raw.info['sfreq'])
+    data, times = raw.get_data(return_times=True)
+    for row, channel_values in enumerate(data):
+        for time, value in zip(times, channel_values):
+            if str(value) == 'nan':
+                print("nan @ %d" % time)
+                onsets.append(time)
+                durations.append(sampling_duration)
+                desc.append('nan')
```

Comment:
It's a typo in regard of the docstring, but is it the good terminology ?

Reply:
> 👍 to annotate all channels (we can't do it any other way anyway), but let's use the time intervals defined by the NaNs. Users can expand these themselves if they want.

Okay so this is what I did: annotating just for the duration of one measurement.

By the way, currently I made the method to annotate one NaN at a time, even if there are several that follow each other. I don't know if it's a good or a bad thing ? It is at least a simple one. I'll just take care of not making duplication if two channels have recorded a NaN value at the same time

Reply:
> By the way, currently I made the method to annotate one NaN at a time, even if there are several that follow each other. I don't know if it's a good or a bad thing ?

Bad thing performance wise, you end up with many more annotations this way. Better to compute where there are contiguous segments and annotate those directly